### PR TITLE
Goblin Ethnicity

### PR DESCRIPTION
**Pull request by ercarp**
## Changelog:
- Created a custom ethnicity for Goblins using `darkspear_trollish` ethnicity as a template.
- Added a set of hairstyles for Goblins.
- Added localization for Goblin and Gilblin ethnicities.

# How to test:
There are two ways you can test this:
1. Open up Ruler Designer and generate male and female Goblins to see what they look like. The ethnicities are currently called **_Goblin_** and **_Gilblin_**.
2. Alternatively, load up into a game and look for Goblins (there are no Gilblins in the game yet) in-game by using the **_Character Finder_** panel.

Provide feedback on the following:
1. The overall look of the ethnicity.
2. The set of possible hairstyles for goblins. _Are there any that should be added or removed?_

❗ **Important**: The Gilblin ethnicity is missing 3D assets integral to their look, such as custom ears and webbed fingers. But since none currently exist in the available bookmarks, it's not a priority to add them anytime soon. That said, I've done my best to give them a decent placeholder appearance.

### CODE REVIEW

**Comment by ercarp** [comment]
A few in-game screenshots for those unable to test on their computers at the moment:

<details><summary>Click to expand</summary>

image
image
image
image
image
image

</details>

**Review by TheDrPibb1999 — APPROVED** [review]
I think the goblins are well made! the jaw is spot on and the fat noses are nice, also the random fat ones caught me off guard and got a good laugh out of me.
Capture

As you said for the gilblin you don't have the 3d models that give them their unique features but their colors and appearance besides those are nice for placeholders until those are added. good work :)

**Comment by ercarp** [comment]
> 
> 
> I think the goblins are well made! the jaw is spot on and the fat noses are nice, also the random fat ones caught me off guard and got a good laugh out of me.

I added a _veeeeeeeeery_ tiny chance for fat goblins to spawn. I called it "Gallywix Gene" in the ethnicity file.

**Comment by TheDrPibb1999** [comment]
> > I think the goblins are well made! the jaw is spot on and the fat noses are nice, also the random fat ones caught me off guard and got a good laugh out of me.
> 
> I added a _veeeeeeeeery_ tiny chance for fat goblins to spawn. I called it "Gallywix Gene" in the ethnicity file.

B E A U T I F U L

File: common/culture/cultures/goblin.txt [hunk 1]
```diff
@@ -55,10 +55,7 @@
 		mother_name_chance = 5
```
Comment (zumbak04) [inline common/culture/cultures/goblin.txt:55]: Mostly goblins look good in western clothes, but female common headgears don't fit them.
<details>
<summary>Click to expand</summary>

image

</details>

File: common/ethnicities/wc_ethnicities_goblin.txt [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+		1 = {  name = forehead_wrinkles_03             range = { 0.1 1 }     }
+	}
+	complexion = {
+        1 = {  name = complexion_1             range = { 0.43 0.43 }     }
+		1 = {  name = complexion_3             range = { 0.43 0.43 }     }
+		1 = {  name = complexion_4             range = { 0.43 0.43 }     }
+		1 = {  name = complexion_5             range = { 0.43 0.43 }     }
+		1 = {  name = complexion_6             range = { 0.43 0.43 }     }
+		1 = {  name = complexion_7             range = { 0.43 0.43 }     }
+	}
```
Comment (zumbak04) [inline common/ethnicities/wc_ethnicities_goblin.txt:419]: Assuming goblins didn't invent 💄.

Comment (ercarp) [reply common/ethnicities/wc_ethnicities_goblin.txt]: Oops. Fixed in 8939248fc1af428a97f830da0133c7a57ea09641.

File: common/culture/cultures/goblin.txt (hunk 1 above)
Comment (ercarp) [reply common/culture/cultures/goblin.txt:55]: > 
> 
> Mostly goblins look good in western clothes, but female common headgears don't fit them.
> Click to expand

I haven't made any changes to their clothing yet, they're just using whatever was assigned to their culture beforehand. That said, noted.

Comment (ercarp) [reply common/culture/cultures/goblin.txt:56]: Fixed in 3a33cac785cac51ae251a808b1b0251bf7ecc61b!

File: common/genes/02_genes_accessories_hairstyles.txt [hunk 3]
```diff
@@ -518,5 +518,43 @@ accessory_genes = {
...
+				1 = male_hair_mena_01 #Curly
+				1 = male_hair_mena_02 #Medium Curls
+				1 = male_hair_mena_04 #Curly Chevelure
+				1 = male_hair_byzantine_01 #Curly Bowl
+				1 = male_hair_byzantine_02 #Roman Fade
+				1 = male_hair_indian_02 #Pulled Back Medium Wavy
+				1 = male_hair_indian_03 #High Bun
+				1 = male_hair_northern_01 #Very Long Wild
+				1 = male_hair_northern_02 #Long Wild
+				1 = male_hair_northern_02
```
Comment (zumbak04) [inline common/genes/02_genes_accessories_hairstyles.txt:536]: The same hairstyle twice.

File: common/genes/02_genes_accessories_hairstyles.txt [hunk 4]
```diff
@@ -518,5 +518,43 @@ accessory_genes = {
...
+				1 = male_hair_steppe_01 #Hair Loops
+				1 = male_hair_steppe_02 #Steppe Rider
```
Comment (zumbak04) [inline common/genes/02_genes_accessories_hairstyles.txt:538]: I would say these hairstyles give barbaric vibes. Goblins aren't barbaric.

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: common/genes/02_genes_accessories_hairstyles.txt [hunk 5]
```diff
@@ -518,5 +518,43 @@ accessory_genes = {
...
+			}		   
+			female = {
+				1 = female_hair_western_03 #Bun
+				1 = female_hair_western_04 #Very Long Double Braids
+				1 = female_hair_western_05 #Medium Wavy
+				1 = female_hair_western_07 #Fish Tails
+				1 = female_hair_western_08 #Double Buns
```
Comment (zumbak04) [inline common/genes/02_genes_accessories_hairstyles.txt:545]: I feel like it's too royal and Medieval for the goblins.
<details>
<summary>Click to expand</summary>

image

</details>

Comment (ercarp) [reply common/genes/02_genes_accessories_hairstyles.txt]: Removed in bf8de12a3c69ce1f80cfda13805fdb11da568491.

File: common/genes/02_genes_accessories_hairstyles.txt (hunk 4 above)
Comment (ercarp) [reply common/genes/02_genes_accessories_hairstyles.txt]: Removed in bf8de12a3c69ce1f80cfda13805fdb11da568491.

File: common/genes/02_genes_accessories_hairstyles.txt (hunk 3 above)
Comment (ercarp) [reply common/genes/02_genes_accessories_hairstyles.txt]: Removed in bf8de12a3c69ce1f80cfda13805fdb11da568491.

**Review by zumbak04 — APPROVED** [review]
LGTM!

**Pull request merged**